src/nvim/api/win_config.c

```diff
@@ -191,6 +191,9 @@ Window nvim_open_win(Buffer buffer, Boolean enter, Dict(float_config) *config, E
   // autocmds in win_enter or win_set_buf below may close the window
   if (win_valid(wp) && buffer > 0) {
     win_set_buf(wp, buf, fconfig.noautocmd, err);
+    if (!fconfig.noautocmd) {
+      apply_autocmds(EVENT_WINNEW, NULL, NULL, false, buf);
```

Comment:
Low-level steps should rarely be done directly in the API function body. Shouldn't this be done in `win_set_buf`? also why doesn't `win_set_buf` already trigger this somehow, e.g. via `win_enter` ?

Reply:
`win_enter` need a flag to trigger `WINNEW` but in `win_new_float` we just check `sync` not set flag `WEE_TRIGGER_NEW_AUTOCMDS` in `win_enter`. let me do some change then..